WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js

```diff
@@ -2849,3 +2856,35 @@ function(check) {
 ZmMailMsgView.prototype._getIframeTitle = function() {
 	return AjxMessageFormat.format(ZmMsg.messageTitle, this._msg.subject);
 };
+
+/**
+ * Setup listeners for mail addresses expand/collapse feature
+ */
+ZmMailMsgView.prototype._setupMailAdressExpandCollapse = function() {
+	this._addressInfoExpanded = Dwt.byId(this._hdrTableId + "_info_expanded");
+	this._addressInfoCollapsed = Dwt.byId(this._hdrTableId + "_info_collapsed");
+	this._addressInfoExpandBtn = Dwt.byId(this._hdrTableId + "_info_expand_btn");
+	this._addressInfoCollapseBtn = Dwt.byId(this._hdrTableId + "_info_collapse_btn");
+	//setup listeners
+	Dwt.setHandler(this._addressInfoExpandBtn, DwtEvent.ONCLICK, this._expandMailAdressContainer.bind(this, false));
+	Dwt.setHandler(this._addressInfoCollapseBtn, DwtEvent.ONCLICK, this._expandMailAdressContainer.bind(this, true));
+	//collapse address container initially
+	this._expandMailAdressContainer(true);
+}
+
+
+/**
+ * @description Expand/collapse detailed address container in mail header
+ * @param collapse: true | false
+ * If collapse = true, collapse the widget else expant it
+ */
+ZmMailMsgView.prototype._expandMailAdressContainer = function(collapse, event) {
+	collapse = !!collapse; //casting to boolean
+	if(collapse) {
```

Comment:
instead of if else we can simplify code like this

this._addressInfoExpanded && Dwt.setVisible(this._addressInfoExpanded, collapse !== true);
this._addressInfoCollapsed && Dwt.setVisible(this._addressInfoCollapsed, collapse === true);